code/src/App.js

```diff
@@ -1,9 +1,92 @@
-import React from 'react';
+/* eslint-disable react/jsx-closing-bracket-location */
+/* eslint-disable comma-dangle */
+import React, { useState, useEffect } from 'react';
+import { ThoughtsList } from 'components/ThoughtsList';
+import { ThoughtsForm } from 'components/ThoughtsForm';
+import { MainLoading } from 'components/MainLoading';
 
+// ///////////////////////////////// STATE  VARIABLES & THEIR SETTERS ///////////////////////// //
+// Declaring variables and setting states
+/* Initializes a state variable thoughts with an empty array value using the useState hook. */
 export const App = () => {
+  const [thoughts, setThoughts] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [newThought, setNewThought] = useState('');
+
+  // calling API to get messages from the server in the JSON format
+  // when API is delayed setLoading will show message
+  const fetchThoughts = () => {
+    setLoading(true);
+    fetch('https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts')
+      .then((response) => response.json())
+      .then((data) => setThoughts(data))
+      .catch((error) => console.log(error))
+      .finally(() => setLoading(false));
+  };
+
+  // Using useEffect() - calling the function fetchThoughts when the component mounts
+  useEffect(() => {
+    fetchThoughts();
+  }, []);
+  // [] dependancy arrey when empty only called when component is mounted
+
```

Comment:
Great commenting, it makes it really easy to read and understand the thought process behind your code!